PWGCF/Tasks/dptdptcorrelations-simchl.cxx

```diff
@@ -1470,40 +1470,188 @@ struct TracksAndEventClassificationQAGen {
   }
 };
 
+/// \brief Checks the correspondence generator level <=> detector level
+struct CheckGeneratorLevelVsDetectorLevel {
+  OutputObj<TH1F> multirecDeltaEta{TH1F("mrDeltaEta", "#Delta#eta multirec tracks;#Delta#eta", 100, -2, 2)};
+  OutputObj<TH1F> multirecDeltaPhi{TH1F("mrDeltaPhi", "#Delta#varphi multirec tracks;#Delta#varphi", 100, 0, 2.0f * static_cast<float>(M_PI))};
+  OutputObj<TH1F> multirecDeltaPt{TH1F("mrDeltaPt", "#Delta#it{p}_{T} multirec tracks;#Delta#it{p}_{T}", 100, 0, 4)};
+  OutputObj<TH1F> multirecDistribution{TH1F("multirec", "Multiple reconstruction; ##/particle; counts", 11, -0.5f, 10.5f)};
+  OutputObj<TH1F> multirecDeltaEta_nc{TH1F("mrDeltaEta_nc", "#Delta#eta multirec tracks;#Delta#eta", 100, -2, 2)};
+  OutputObj<TH1F> multirecDeltaPhi_nc{TH1F("mrDeltaPhi_nc", "#Delta#varphi multirec tracks;#Delta#varphi", 100, 0, 2.0f * static_cast<float>(M_PI))};
+  OutputObj<TH1F> multirecDeltaPt_nc{TH1F("mrDeltaPt_nc", "#Delta#it{p}_{T} multirec tracks;#Delta#it{p}_{T}", 100, 0, 4)};
+  OutputObj<TH1F> multirecDistribution_nc{TH1F("multirec_nc", "Multiple reconstruction; ##/particle; counts", 11, -0.5f, 10.5f)};
+  TDatabasePDG* fPDG;
+
+  void init(InitContext const& context)
+  {
+    fPDG = TDatabasePDG::Instance();
+  }
+
+  void process(soa::Join<aod::Tracks, aod::McTrackLabels> const& tracks, aod::McParticles const& mcParticles)
+  {
+    constexpr float TWOPI = 2.0F * static_cast<float>(M_PI);
+
+    std::vector<std::vector<int64_t>> mclabelpos(mcParticles.size());
+    std::vector<std::vector<int64_t>> mclabelneg(mcParticles.size());
+    std::vector<std::vector<int64_t>> mclabelpos_negcoll(mcParticles.size());
+    std::vector<std::vector<int64_t>> mclabelneg_negcoll(mcParticles.size());
+
+    size_t nreco = tracks.size();
+    size_t ngen = 0;
+
+    for (auto& part : mcParticles) {
+      auto pdgpart = fPDG->GetParticle(part.pdgCode());
+      if (pdgpart != nullptr) {
+        float charge = (pdgpart->Charge() / 3 >= 1) ? 1.0 : ((pdgpart->Charge() / 3 <= -1) ? -1.0 : 0.0);
```

Comment:
Why divide by 3 before comparing to 1 and -1 ?

Reply:
An acquired routine when working with PDG
I will substitute it with `>= 3` and `<= -3`

Reply:
Or just pdgpart->Charge != 0 given your logic? :)

Reply:
Well, we don't want the fractional charge particles be accounted

Reply:
But isPhysicalPrimary should never accept any of those...